### PR TITLE
add reusable AWS module

### PR DESCRIPTION
**Pull request by DaMandal0rian**
This is part 1 of refactoring the terraform modules to be more composable. The PR adds a common templates directory and AWS modules.

- adds the reusable aws modules for (VPC and EC2) resources

closes #142

### CODE REVIEW

File: templates/terraform/aws/ec2/ami.tf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+
+  owners = ["099720109477"]
```
Comment (vedhavyas) [inline templates/terraform/aws/ec2/ami.tf:19]: Can't we move this as a variable instead ?

File: templates/terraform/aws/vpc/main.tf [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+    vpc_id = try(aws_vpc_ipv4_cidr_block_association.this[0].vpc_id, aws_vpc.this[0].id, "")
+
+    create_vpc = var.create_vpc
+  }
+
+  ######################################
+  # VPC
+  ######################################
+
+  resource "aws_vpc" "this" {
```
Comment (vedhavyas) [inline templates/terraform/aws/vpc/main.tf:21]: I dont think `this` is generally preferred. It would be ideal to name this instead

**Review by vedhavyas — COMMENTED** [review]
I see only templates in this PR
Where are we planning to use this module ?

File: templates/terraform/aws/ec2/ami.tf (hunk 1 above)
Comment (DaMandal0rian) [reply templates/terraform/aws/ec2/ami.tf:19]: This is the UID for canonical which is constant.

File: templates/terraform/aws/vpc/main.tf (hunk 2 above)
Comment (DaMandal0rian) [reply templates/terraform/aws/vpc/main.tf:21]: Please see https://www.terraform-best-practices.com/naming

"this" or "main" is usually used in naming convention if there is one resource created in the module.

**Comment by DaMandal0rian** [comment]
> I see only templates in this PR
> Where are we planning to use this module ?

This is just a PR for the module, it will be used by all resources that we provision on AWS, which is being refactored and will be in a future PR. If I include it here, there will be many files :)

**Review by vedhavyas — APPROVED** [review]

**Pull request merged**